engine/engine.go

```diff
@@ -358,8 +370,23 @@ func (q *Query) Explain() *ExplainOutputNode {
 	return explainVector(q.exec)
 }
 
-func (q *Query) Profile() {
-	// TODO(bwplotka): Return profile.
+func (q *Query) Analyze() *AnalyzeOutputNode {
+
+	return analyzeVector(q.exec.(model.ObservableVectorOperator))
```

Comment:
This will crash if the type assertion fails. Perhaps we could return `nil` if this fails?

```go
observableRoot, ok := q.exec.(model.ObservableVectorOperator)
if !ok { return nil }
...
```